pyterrier/pipelines.py

```diff
@@ -186,6 +202,11 @@ def _run_and_evaluate(
                 raise ValueError("batch of %d topics, but no results received in batch %d from %s" % (len(batch_topics), i, str(system) ) )
             endtime = timer()
             runtime += (endtime - starttime) * 1000.
+
+            # write results to save_file; we will append for subsequent batches
+            if save_file is not None:
```

Comment:
should this file be cleaned up if an exception is thrown? Otherwise, the partial file will still be around and picked up on a subsequent execution (mistakenly thinking it's a complete file).

Reply:
Also, won't appending cause problems for gzip files?

Reply:
Yes, a catch clause should be added to remove the file.

I can confirm that GZip files can be appended with gzip files.